lib/mix/lib/mix/dep/lock.ex

```diff
@@ -62,4 +63,11 @@ defmodule Mix.Dep.Lock do
   defp lockfile do
     Mix.Project.config[:lockfile]
   end
+
+  defp assert_no_merge_conflicts_in_lockfile(lockfile, info) do
+    if info =~ ~r/^(?:<<<<<<<|=======|>>>>>>>|\|\|\|\|\|\|\|)/ do
```

Comment:
I think it should be enough, but bundler was using this regex, so I am not sure if there is some edge case with mercurial, bazaar or some other VCS where there could be only one of these.

> `match?({0, _}, :binary.match(info, ["<<<", "===", ">>>", ...]))` will match only on the front. You can also do normal binary pattern matching:

Thank you, you just made me realize there was a bug in the current code, the regex should be `~r/^(?:<<<<<<<|=======|>>>>>>>|\|\|\|\|\|\|\|)/m`, without the `m` modifier the `^` does not behave as intended.
I think your proposals would work without the `m` modifier, but with it, we would need to split the string right?